gratipay/utils/__init__.py

```diff
@@ -132,6 +133,46 @@ def get_team(state):
     return team
 
 
+def b64encode_s(s):
+    prefix = b''
+    if not isinstance(s, bytes):
+        s = s.encode('utf8')
+    else:
+        # Check whether the string is binary or already utf8
+        try:
+            s.decode('utf8')
+        except UnicodeError:
+            prefix = b'.'
+    return prefix + b64encode(s, b'-_').replace(b'=', b'~')
+
+
+def b64decode_s(s, **kw):
+    def error():
+        if 'default' in kw:
+            return kw['default']
+        raise Response(400, "invalid base64 input")
+
+    try:
+        s = s.encode('ascii')
+    except UnicodeError:
+        return error()
+
+    udecode = lambda a: a.decode('utf8')
+    if s[:1] == b'.':
+        udecode = lambda a: a
+        s = s[1:]
+    s = s.replace(b'~', b'=')
+    try:
+        return udecode(b64decode(s, '-_'))
+    except Exception:
+        try:
+            # For retrocompatibility
+            return udecode(b64decode(s))
+        except Exception:
+            pass
+        return error()
+
+
```

Comment:
> Safe against bugs in ASP.NET MVC

It could happen with [jquery](http://stackoverflow.com/questions/7235874/jquery-query-parameter-parsing-in-query-string-breaks-query-string-up) for all we know. ;)

> Why can't the caller try/except for Response?

Hm. Good idea. We could do that given that this was indeed @Changaco's intention for using `default`. Is that right @Changaco ?